From 51348427e294a9d6d09b001c915c7a14f4f9e416 Mon Sep 17 00:00:00 2001
From: MowFord <131182600+MowFord@users.noreply.github.com>
Date: Fri, 15 Dec 2023 16:30:25 -0600
Subject: [PATCH] Monk mob weapon damage adjustments to match PC

---
 src/map/utils/mobutils.cpp | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/src/map/utils/mobutils.cpp b/src/map/utils/mobutils.cpp
index 1b0397426e9..75384619e87 100644
--- a/src/map/utils/mobutils.cpp
+++ b/src/map/utils/mobutils.cpp
@@ -56,6 +56,7 @@ namespace mobutils
     uint16 GetWeaponDamage(CMobEntity* PMob, uint16 slot)
     {
         uint16 lvl    = PMob->GetMLevel();
+        auto*  weapon = dynamic_cast<CItemWeapon*>(PMob->m_Weapons[SLOT_MAIN]);
         int8   bonus  = 2;
         uint16 damage = 0;
 
@@ -69,7 +70,17 @@ namespace mobutils
             bonus = 0;
         }
 
-        damage = lvl + bonus;
+        if (slot == SLOT_MAIN && (weapon == nullptr || weapon->getSkillType() == SKILL_HAND_TO_HAND))
+        {
+            // https://ffxiclopedia.fandom.com/wiki/Category:Hand-to-Hand
+            // base h2h weapon dmg for a given h2h skill: (0.11 * h2h skill) + 3
+            uint16 h2hskill = battleutils::GetMaxSkill(SKILL_HAND_TO_HAND, JOB_MNK, lvl);
+            damage          = 0.11f * h2hskill + 3;
+        }
+        else
+        {
+            damage = lvl + bonus;
+        }
 
         damage = (uint16)(damage * PMob->m_dmgMult / 100.0f);
 
