diff --git a/src/map/ai/states/attack_state.cpp b/src/map/ai/states/attack_state.cpp
index 717264fc68..554b6da1af 100644
--- a/src/map/ai/states/attack_state.cpp
+++ b/src/map/ai/states/attack_state.cpp
@@ -125,8 +125,8 @@ void CAttackState::UpdateTarget(uint16 targid)
             {
                 for (auto&& PPotentialTarget : PChar->SpawnMOBList)
                 {
-                    if (PPotentialTarget.second->animation == ANIMATION_ATTACK && facing(PChar->loc.p, PPotentialTarget.second->loc.p, 64) &&
-                        distance(PChar->loc.p, PPotentialTarget.second->loc.p) <= 10)
+                    if (PPotentialTarget.second->animation == ANIMATION_ATTACK &&
+                        distance(PChar->loc.p, PPotentialTarget.second->loc.p) <= 20.0f)
                     {
                         std::unique_ptr<CBasicPacket> errMsg;
                         if (PChar->IsValidTarget(PPotentialTarget.second->targid, TARGET_ENEMY, errMsg))
